### PR TITLE
feat(parallel): add a semaphore to limit maximum connections

### LINKED ISSUE #10: define a maximum concurrency level
We currently spawn as many things concurrently as we can, and as seen in #8 this can cause 3-stooges syndrome where tasks start timing out even though each one individually could easily resolve in time. We should have some upper limit on concurrency to manage this, or... do something more robust.

### PR DESCRIPTION
**Pull request by Gankra**
fixes #10

### CODE REVIEW

**Comment by Gankra** [comment]
probably this PR should add backoff/retry semantics but idk how to do that yet

**Review by shadows-withal — APPROVED** [review]

**Comment by Gankra** [comment]
https://docs.rs/backon/latest/backon/

**Review by mistydemeo — APPROVED** [review]

**Comment by Gankra** [comment]
Added retrying with these default semantics on 500's https://docs.rs/backon/latest/backon/struct.ExponentialBuilder.html

File: Cargo.toml [hunk 1]
```diff
@@ -33,3 +33,4 @@ reqwest = { version = "0.11.22", default-features = false, optional = true, feat
     "rustls-tls",
     "json",
 ]}
+backon = "0.4.1"
```
Comment (mistydemeo) [inline Cargo.toml:36]: Should this be optional so we don't use it in faux-schema mode?

**Pull request merged**